src/utils/index.js

```diff
@@ -125,10 +125,13 @@ function deepEqual (a, b) {
   var keysB;
   var valA;
   var valB;
+  var areBothObjects = a && b && a.constructor === Object && b.constructor === Object;
```

Comment:
Should we evaluate these inline so that the if expression can lazy evaluate and possibly not have to do `a && b` check twice?